### ISSUE TITLE
Costco receipt graphql error

### ISSUE THREAD

**Issue by drewvolz** [ankurdave/beancount_import_sources#1]
Hey @ankurdave, running [`download_costco_receipts.js`](https://github.com/ankurdave/beancount_import_sources/blob/main/download/download_costco_receipts.js) seems to be returning an error. 

```sh
Status: 400 TARGET_RESP_FLOW - General processing error
```
```json
{
  "context": {
    "statusMessage": {
      "statusCode": "APIGATEWAY.BACKEND.FAULT",
      "errorCode": "400",
      "descriptions": [
        {
          "value": "Received non success response code||errorType:ClientError|errorMessage:[Error trying to resolve field 'receipts'. Object reference not set to an instance of an object.]|errorCode:-2147467261|source:Order|trackingId:rrt-06ef473972efcec74-c-wo-20225-15840855-1",
          "localeCode": "en_US"
        }
      ]
    },
    "messageName": "TARGET_RESP_FLOW|null|null|null|messaging.adaptors.http.flow.ErrorResponseCode|0|0|1|true",
    "timestamp": "2023-11-01T03:29:06.540Z",
    "sendingApplication": "APIgee Edge Server",
    "trackingId": "rrt-06ef473972efcec74-c-wo-20225-15840855-1"
  }
}
```

I'm thinking that maybe the graphql query has changed on Costco's end? When I look at successful 200 responses, I'm seeing the following payload:

```json
  "variables": {
    "startDate": "5/01/2023",
    "endDate": "10/31/2023",
    "text": "Last 6 Months"
  }
```

```graphql
query receipts($startDate: String!, $endDate: String!) {
  receipts(startDate: $startDate, endDate: $endDate) {
    warehouseName
    documentType
    transactionDateTime
    transactionBarcode
    warehouseName
    transactionType
    total
    totalItemCount
    itemArray {
      itemNumber
    }
    tenderArray {
      tenderTypeCode
      tenderDescription
      amountTender
    }
    couponArray {
      upcnumberCoupon
    }
  }
}
```

**Comment by ankurdave** [comment]
I've started seeing this error as well. I had started looking into it a few days ago, but wasn't able to get it working yet. Looking at the network requests that the Costco receipt pages make, the GraphQL endpoint and authentication mechanisms also seem to have changed. Authentication now seems to happen via a request to https://signin.costco.com/e0714dd4-784d-46d6-a278-3e29553483eb/b2c_1a_sso_wcs_signup_signin_567/oauth2/v2.0/token, and the GraphQL endpoint now seems to be https://ecom-api.costco.com/ebusiness/product/v1/products/graphql.
